### PR TITLE
[ABW-3756] Add preferences for AuthorizedDapp

### PR DESCRIPTION
**Pull request by matiasbzurovski**
## Changelog
- Allows configuring preferences for authorized dapps. For now, only configuration available is to show/hide direct deposit claims for the given dapp.

[JIRA Ticket](https://radixdlt.atlassian.net/browse/ABW-3756)

### CODE REVIEW

**Review by CyonAlexRDX — APPROVED** [review]
LGTM

File: apple/Sources/Sargon/Extensions/Methods/Profile/AuthorizedDapp+Wrap+Functions.swift [hunk 1]
```diff
@@ -17,4 +17,12 @@ extension AuthorizedDapp {
 	public func jsonData() -> Data {
 		authorizedDappToJsonBytes(authorizedDapp: self)
 	}
+	
+	public mutating func showDeposits(_ show: Bool) {
+		preferences.deposits = show ? .visible : .hidden
+	}
+	
+	public var showDeposits: Bool {
```
Comment (GhenadieVP) [inline apple/Sources/Sargon/Extensions/Methods/Profile/AuthorizedDapp+Wrap+Functions.swift:25]: maybe follow the usual guideline of using `is` prefix.

**Review by GhenadieVP — APPROVED** [review]

File: apple/Sources/Sargon/Extensions/Methods/Profile/AuthorizedDapp+Wrap+Functions.swift (hunk 1 above)
Comment (matiasbzurovski) [reply apple/Sources/Sargon/Extensions/Methods/Profile/AuthorizedDapp+Wrap+Functions.swift:25]: `isDepositsVisible`?

Comment (GhenadieVP) [reply apple/Sources/Sargon/Extensions/Methods/Profile/AuthorizedDapp+Wrap+Functions.swift:25]: something like that.

File: src/profile/supporting_types/authorized_dapp_detailed.rs [hunk 2]
```diff
@@ -23,6 +23,9 @@ pub struct AuthorizedDappDetailed {
     pub display_name: Option<DisplayName>,
 
     pub detailed_authorized_personas: DetailedAuthorizedPersonas,
+
+    #[serde(default)]
```
Comment (CyonAlexRDX) [inline src/profile/supporting_types/authorized_dapp_detailed.rs]: @matiasbzurovski I think you can remove 

```
    Serialize,
    Deserialize,
``` 
from `AuthorizedDappDetailed`

and also all `#serde` ofc... It seems it is not needed! does that compile?

Comment (matiasbzurovski) [reply src/profile/supporting_types/authorized_dapp_detailed.rs]: right, that makes more sense based on our Slack discussion. Removed

**Pull request merged**